test/test_pipeline/components/data_preprocessing/test_data_preprocessing_categorical.py

```diff
@@ -0,0 +1,71 @@
+import unittest
+import numpy as np
+from scipy import sparse
+
+from autosklearn.pipeline.components.data_preprocessing.data_preprocessing_categorical \
+    import CategoricalPreprocessingPipeline
+
+
+class CategoricalPreprocessingPipelineTest(unittest.TestCase):
+
+    def test_fit_transform(self):
+        X = np.array([
+            [1, 2, 0],
+            [3, 0, 0],
+            [2, 9, np.nan]])
+        Y = np.array([
+            [1, 0, 0, 0, 1, 0, 0, 1],
+            [0, 0, 1, 1, 0, 0, 0, 1],
+            [0, 1, 0, 0, 0, 1, 1, 0]])
+        # dense input
+        Yt = CategoricalPreprocessingPipeline().fit_transform(X)
+        self.assertTrue((Yt.todense() == Y).all())
+        # sparse input
+        X_sparse = sparse.csc_matrix(X+1)
```

Comment:
Because X is dense matrix containing zeros as valid categories. If I convert X to sparse, those categories will disappear. That is why I'm converting X+1 instead. 
But to avoid confusion, I'll edit the code and use an X without zeros.